### PR TITLE
Use configurable sub loggers for different components

### PR DESCRIPTION
**Pull request by ChuntaoLu**
This PR addresses https://github.com/uber/zanzibar/issues/275.

### CODE REVIEW

File: runtime/gateway.go [hunk 1]
```diff
@@ -443,9 +443,10 @@ func (gateway *Gateway) initJaegerConfig(config *StaticConfig) *jaegerConfig.Con
 }
 
 func (gateway *Gateway) setupTracer(config *StaticConfig) error {
+	infoToDebug := config.MustGetBoolean("tchannel.infoToDebug")
 	opts := []jaegerConfig.Option{
 		// TChannel logger implements jaeger logger interface
-		jaegerConfig.Logger(NewTChannelLogger(gateway.Logger)),
+		jaegerConfig.Logger(NewTChannelLogger(gateway.Logger, infoToDebug)),
```
Comment (Raynos) [inline runtime/gateway.go]: its weird to downgrade jaeger here.

File: runtime/gateway.go [hunk 2]
```diff
@@ -443,9 +443,10 @@ func (gateway *Gateway) initJaegerConfig(config *StaticConfig) *jaegerConfig.Con
 }
 
 func (gateway *Gateway) setupTracer(config *StaticConfig) error {
+	infoToDebug := config.MustGetBoolean("tchannel.infoToDebug")
```
Comment (Raynos) [inline runtime/gateway.go]: Use "jaeger.infoToDebug"

File: runtime/gateway.go [hunk 3]
```diff
@@ -407,16 +420,21 @@ func (gateway *Gateway) setupLogger(config *StaticConfig) error {
...
 	)
 	zapLogger := zap.New(
 		NewInstrumentedZapCore(
 			prodCore, gateway.AllHostScope,
 		),
 	)
 
+	gateway.logEncoder = logEncoder
+	if gateway.logWriter == nil {
+		gateway.logWriter = output
```
Comment (Raynos) [inline runtime/gateway.go]: need to always assign this to `output`

In fact; let's not re-use the `logWriter` field and just add a new top level field for `gateway.logWriteSyncer` or something

Comment (ChuntaoLu) [reply runtime/gateway.go]: Done. Curious why do you think it's better to not reuse `logWriter`?

**Pull request merged**